packages/components/src/components/drawer/drawer.ts

```diff
@@ -200,6 +200,11 @@ export default class SdDrawer extends SolidElement {
         animateTo(this.overlay, overlayAnimation.keyframes, overlayAnimation.options)
       ]);
 
+      //Add a11y attributes to close button
+      const closeButton = this.shadowRoot?.getElementById('close-button')?.shadowRoot?.querySelector('button');
```

Comment:
I feel this should go to a separate getter function, to avoid duplicated code on `line 250`.

Reply:
https://lit.dev/docs/components/shadow-dom/#query

Please use this.

Reply:
Why can't you set this directly on the element that it's done reactively?

Reply:
Updated